### ISSUE TITLE
Enhancement: label support for input-time-picker

### ISSUE THREAD

**Issue by eriklharper** [Esri/calcite-components#1759]
### Description

This issue will add full labelling support for `calcite-input-time-picker`.

### Acceptance Criteria 

`calcite-input-time-picker` should support wrapping `calcite-label` and `label` elements with tests reflecting this usage pattern.

#### Which Component

`calcite-input-time-picker`

#### Original reference comment

I think I'll create a follow-up issue to address label support, which will remove this aria-label and add labels around the element in the tests.

_Originally posted by @eriklharper in https://github.com/Esri/calcite-components/pull/1736#discussion_r592694064_

**Comment by macandcheese** [comment]
@eriklharper is this still valid?

**Comment by eriklharper** [comment]
> @eriklharper is this still valid?

Yes, still relevant.  We can lump this issue under a "native form support" epic which has been thrown around as a possible effort we might pursue for all the calcite form components. @jcfranco
